From 4b324e1fe4091cd737987e2418f655d9658a324e Mon Sep 17 00:00:00 2001
From: Noah Shutty <noahshutty@gmail.com>
Date: Fri, 30 Oct 2020 17:48:46 -0700
Subject: [PATCH] Fix PostgreSQL tests build under macOS

Use fully qualified "soci::session" type name to avoid conflicts with
"session" struct declared in sys/proc.h under macOS.

This is a recurrent problem (see #605 for a previous example of it)
which should probably be solved by removing "using namespace soci" from
the test to prevent it from happening again.

Closes #832.
---
 tests/postgresql/test-postgresql.cpp | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/tests/postgresql/test-postgresql.cpp b/tests/postgresql/test-postgresql.cpp
index d25606a1..3db1a5bd 100644
--- a/tests/postgresql/test-postgresql.cpp
+++ b/tests/postgresql/test-postgresql.cpp
@@ -994,7 +994,7 @@ TEST_CASE("Bulk iterators", "[postgresql][bulkiters]")
 // false_bind_variable_inside_identifier
 struct test_false_bind_variable_inside_identifier_table_creator : table_creator_base
 {
-    test_false_bind_variable_inside_identifier_table_creator(session & sql)
+    test_false_bind_variable_inside_identifier_table_creator(soci::session & sql)
         : table_creator_base(sql)
         , msession(sql) 
     {
@@ -1028,7 +1028,7 @@ private:
         }
         catch (soci_error const& e){}
     }
-    session& msession;
+    soci::session& msession;
 };
 TEST_CASE("false_bind_variable_inside_identifier", "[postgresql][bind-variables]")
 {
@@ -1037,7 +1037,7 @@ TEST_CASE("false_bind_variable_inside_identifier", "[postgresql][bind-variables]
     std::string type_value;
 
     {
-        session sql(backEnd, connectString);
+        soci::session sql(backEnd, connectString);
         test_false_bind_variable_inside_identifier_table_creator tableCreator(sql);
 
         sql << "insert into soci_test(\"column_with:colon\") values(2020)";
-- 
2.23.0

